client/my-sites/plans-features-main/hooks/use-intent-from-site-meta.ts

```diff
@@ -0,0 +1,35 @@
+import { useSiteIntent } from '@automattic/data-stores';
+import { useSelector } from 'react-redux';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import type { Intent } from 'calypso/my-sites/plans-features-main/hooks/use-plan-types-with-intent';
+
+interface IntentFromSiteMeta {
+	processing: boolean;
```

Comment:
Nit: I was wondering if we can use the default `isFetching` from react-query instead of a custom `processing`.

Reply:
I want "processing" (and the whole layer in this hook) to be an abstraction, for now. No errors or react-query specific things propagating further. Along with an `undefined` intent, it's enough to continue processing and fail gracefully (or pick up once defined). 

It could be something to engineer more down the line.